lisa/sut_orchestrator/azure/features.py

```diff
@@ -2151,6 +2162,135 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
                     )
 
 
+class Availability(AzureFeatureMixin, features.Availability):
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+        self._initialize_information(self._node)
+
+    @classmethod
+    def settings_type(cls) -> Type[schema.FeatureSettings]:
+        return AvailabilitySettings
+
+    @classmethod
+    def create_setting(
+        cls, *args: Any, **kwargs: Any
+    ) -> Optional[schema.FeatureSettings]:
+        raw_capabilities: Any = kwargs.get("raw_capabilities")
+        availability_settings: AvailabilitySettings = AvailabilitySettings()
+        availability_settings.availability_type = search_space.SetSpace(
+            True,
+            [
+                AvailabilityType.NoRedundancy,
+                AvailabilityType.AvailabilitySet,
+            ],
+        )
+
+        availability_zones = raw_capabilities.get("availability_zones", None)
+        if availability_zones:
+            availability_settings.availability_type.add(
+                AvailabilityType.AvailabilityZone
+            )
+            availability_settings.availability_zones = search_space.SetSpace(
+                is_allow_set=True, items=availability_zones
+            )
+
+        return availability_settings
+
+    @classmethod
+    def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
+        environment = cast(Environment, kwargs.get("environment"))
+        arm_parameters = cast(AzureArmParameter, kwargs.get("arm_parameters"))
+        params = arm_parameters.availability_options
+
+        is_ultra_disk = any(
+            [
+                node.data_disk_type == _disk_type_mapping[schema.DiskType.UltraSSDLRS]
+                for node in arm_parameters.nodes
+            ]
+        )
+        is_maximize_capability = any(
+            [node.maximize_capability for node in arm_parameters.nodes]
+        )
+
+        node_availability_capabilities: List[AvailabilitySettings] = list(
+            environment.nodes.list_features_by_type(AvailabilitySettings)
```

Comment:
Use the feature from first node, it should be good enough now.

Reply:
Is it safe to assume that all the nodes in an envirnment will be the same setup? For example, data_disk_type? It could be the case that one node support availability zones, but not all, right?

Reply:
Yes. We can assume all nodes are the same configurations for now. I tried to support different schemas in the same env, but no scenario and high cost.